File: 20220817/모범답안/20220817_05.js/4Rule.py

def solution(S):
    arr = [ s for s in S ]
    numbers = re.findall(r'\d+', S)
    operators = re.findall(r'[+\-*/]', S)

    temp = float(numbers[0])
    result = 0

    for i in range (len(operators)):
        if operators[i] == '*':
            temp *= float(numbers[i+1])
        elif operators[i] == '/':
            temp /= float(numbers[i+1])
        elif operators[i] == '+':
            result += temp
            temp = 0
            temp += float(numbers[i+1])
        elif operators[i] == '-':
            result += temp
            temp = 0
            temp -= float(numbers[i+1])

    result += temp

    return '%0.2f' % result

File: 20220817/모범답안/20220817_05.js/4Rule.js


function solution(S){
    var arr = S.split("");
    var numbers = S.match(/\d+/gi);
    var operators = S.match(/[+\-*/]/gi);
  
    var temp = parseFloat(numbers[0]);
    var result = 0;
  
    for(var i = 0; i < operators.length; i++){
      switch (operators[i]) {
        case '*':
          temp *= parseFloat(numbers[i+1]);
          break;
        case '/':
          temp /= parseFloat(numbers[i+1]);
          break;
        case '+':
          result += temp;
          temp = 0;
          temp += parseFloat(numbers[i+1]);
          break;
        case '-':
          result += temp;
          temp = 0;
          temp -= parseFloat(numbers[i+1]);
          break;
      }
    }
  
    result += temp;
  
    return result.toFixed(2);
  }